components/authentication-framework/org.wso2.carbon.identity.application.authentication.endpoint.util/src/main/java/org/wso2/carbon/identity/application/authentication/endpoint/util/filter/AuthParameterFilter.java

```diff
@@ -0,0 +1,190 @@
+                HttpServletRequest httpServletRequest = cacheParamsFromErrorKey((HttpServletRequest) servletRequest);
+                filterChain.doFilter(httpServletRequest, servletResponse);
+                return;
+            }
+        }
+        filterChain.doFilter(servletRequest, servletResponse);
+    }
+
+    /**
+     * Cache the parameters retrieved from the API using SessionDataKey.
+     *
+     * @param servletRequest - Received servlet request.
+     * @return - HttpServletRequestWrapper with cached parameter map.
+     */
+    private HttpServletRequest cacheParamsFromSessionDataKey(HttpServletRequest servletRequest) {
+
+        String authAPIURL = buildAPIPath("sessionDataKey", servletRequest.getParameter("sessionDataKey"));
+        String contextProperties = AuthContextAPIClient.getContextProperties(authAPIURL);
+        Gson gson = new Gson();
+        Map<String, Object> cachedParams = gson.fromJson(contextProperties, Map.class);
+
+        return new AuthenticationRequestWrapper(servletRequest, cachedParams);
+    }
+
+    /**
+     * Cache the parameters retrieved from the API using SessionDataKey.
+     *
+     * @param servletRequest - Received servlet request.
+     * @return - HttpServletRequestWrapper with cached parameter map.
+     */
+    private HttpServletRequest cacheParamsFromConsentKey(HttpServletRequest servletRequest) {
+
+        String authAPIURL = buildAPIPath("sessionDataKeyConsent", servletRequest.getParameter("sessionDataKeyConsent"));
+        String contextProperties = AuthContextAPIClient.getContextProperties(authAPIURL);
+        Gson gson = new Gson();
+        Map<String, Object> cachedParams = gson.fromJson(contextProperties, Map.class);
+        return new AuthenticationRequestWrapper(servletRequest, cachedParams);
+    }
+
+    private HttpServletRequest cacheParamsFromErrorKey(HttpServletRequest servletRequest) {
+        String authAPIURL = buildAPIPath("errorKey", servletRequest.getParameter("errorKey"));
+        String contextProperties = AuthContextAPIClient.getContextProperties(authAPIURL);
+        Gson gson = new Gson();
+        Map<String, Object> cachedParams = gson.fromJson(contextProperties, Map.class);
+        return new AuthenticationRequestWrapper(servletRequest, cachedParams);
+    }
+
+    /**
+     * Build the API Path using the key and value.
+     *
+     * @param key   - SessionDataKey or SessionDataKeyConsent.
+     * @param value - Value of the key.
+     * @return - API Path.
+     */
+    private String buildAPIPath(String key, String value) {
+
+        String authAPIURL = context.getInitParameter(Constants.AUTHENTICATION_REST_ENDPOINT_URL);
+
+        if (!StringUtils.isBlank(authAPIURL)) {
+            if (resolveTenantDomain().equals("carbon.super")) {
+                authAPIURL = authAPIURL.replace("t/${tenant}/", "");
+            } else {
+                authAPIURL = authAPIURL.replace("${tenant}", resolveTenantDomain());
+            }
```

Comment:
Why do we need to specially handle this URL?

Reply:
`AUTHENTICATION_REST_ENDPOINT_URL` should be tenanted for the tenanted applications in Asgardeo.